### PR TITLE
feat: use actionCreator for account actions

### PR DESCRIPTION
**Pull request by ReeceHoffmann**
Example implementation of action creation using createAction.

### CODE REVIEW

File: src/js/account/actions.js [hunk 1]
```diff
@@ -46,9 +49,12 @@ export const updateAccount = update => ({
...
-export const updateAccountSettings = update => ({
-    type: UPDATE_ACCOUNT_SETTINGS.REQUESTED,
-    update
+export const updateAccountSettings = createAction(UPDATE_ACCOUNT_SETTINGS.REQUESTED, update => {
+    return {
+        payload: {
+            update
+        }
+    };
 });
```
Comment (igboyes) [inline src/js/account/actions.js]: You can write this without a `return`.

File: src/js/account/actions.js [hunk 2]
```diff
@@ -59,10 +65,13 @@ export const updateAccountSettings = update => ({
  * @param newPassword {string} the new account password
  * @returns {object}
  */
-export const changePassword = (oldPassword, newPassword) => ({
-    type: CHANGE_ACCOUNT_PASSWORD.REQUESTED,
-    oldPassword,
-    newPassword
+export const changePassword = createAction(CHANGE_ACCOUNT_PASSWORD.REQUESTED, (oldPassword, newPassword) => {
+    return {
```
Comment (igboyes) [inline src/js/account/actions.js]: You can write this without a `return`.

File: src/js/account/actions.js [hunk 3]
```diff
@@ -103,10 +110,8 @@ export const clearAPIKey = simpleActionCreator(CLEAR_API_KEY);
  * @param permissions {object} permissions configuration object for the new API key
  * @returns {object}
  */
-export const updateAPIKey = (keyId, permissions) => ({
-    type: UPDATE_API_KEY.REQUESTED,
-    keyId,
-    permissions
+export const updateAPIKey = createAction(UPDATE_API_KEY.REQUESTED, (keyId, permissions) => {
+    return { payload: { keyId, permissions } };
```
Comment (igboyes) [inline src/js/account/actions.js]: You can write this without a `return`.

**Review by igboyes — CHANGES_REQUESTED** [review]
This looks good to me and I think we should continue with it. I actually like the data being in a nested `payload` object.

Please deal with this warning from GitHub actions:
> Check warning on line 142 in src/js/errors/reducer.js

You can pare down a lot of anonymous arrow functions in this PR by eliminating `return` statements where possible. I only commented a few.

**Comment by ReeceHoffmann** [comment]
> This looks good to me and I think we should continue with it. I actually like the data being in a nested `payload` object.
> 
> Please deal with this warning from GitHub actions:
> 
> > Check warning on line 142 in src/js/errors/reducer.js
> 
> You can pare down a lot of anonymous arrow functions in this PR by eliminating `return` statements where possible. I only commented a few.

Sounds good, I will get to work updating the rest of the action creators. I changed all the arrow functions to function as "one line" arrow functions (automatic return) and fixed the oversight that caused that warning.

**Review by igboyes — APPROVED** [review]

**Pull request merged**